docs-ci: `make linkcheck` prone to transient network failures

I've just run into [this error](https://github.com/nextstrain/augur/actions/runs/10530367718/job/29180176987#step:6:731) on an Augur PR which did not change any docs links:

```
(api/developer/augur.merge: line    7) broken    https://www.gnu.org/software/bash/manual/bash.html#ANSI_002dC-Quoting - HTTPSConnectionPool(host='www.gnu.org', port=443): Max retries exceeded with url: /software/bash/manual/bash.html (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f4835cbc1c0>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
build finished with problems.
make: *** [Makefile:20: linkcheck] Error 1
```

This seems like a transient network error which shows up as a failing check ❌ on the PR which confused me at first.

## Possible solutions

1. ⛔️ Split `linkcheck` into a separate job on docs-ci and use `continue-on-error: true`
    - https://github.com/nextstrain/.github/pull/107
    - Not a great solution because it masks real linkcheck issues
3. Don't run `linkcheck` in CI but instead on a weekly schedule
4. Ignore `timeout` codes
    - https://github.com/nextstrain/.github/pull/110
    - This should be a good improvement but does not cover all transient failures (see below)
5. (per-project) Handle `broken` codes on a case-by-case basis. If it's intermittent, add to `linkcheck_ignore`. Otherwise, update or remove the link.

--- a/.github/workflows/docs-ci.yaml
+++ b/.github/workflows/docs-ci.yaml
@@ -46,6 +46,7 @@ env:
   # --keep-going: find all warnings
   # https://www.sphinx-doc.org/en/master/man/sphinx-build.html
   SPHINXOPTS: -n -W --keep-going
+  BUILDDIR: _build
 
 jobs:
   build:
@@ -75,5 +76,16 @@ jobs:
       - run: make ${{ inputs.make-target }}
         working-directory: ${{ inputs.docs-directory }}
 
-      - run: make linkcheck
+      # Ignore the exit code, results will be checked in the next step
+      - run: make linkcheck || true
+        working-directory: ${{ inputs.docs-directory }}
+
+      - name: Check for broken links
+        run: |
+          broken_links=$(jq 'select(.status == "broken")' "$BUILDDIR/linkcheck/output.json")
+          if [ -n "$broken_links" ]; then
+            echo "Broken links found:"
+            echo "$broken_links"
+            exit 1
+          fi
         working-directory: ${{ inputs.docs-directory }}

